pkg/testutil/require.go

```diff
@@ -293,11 +296,90 @@ func RequireDecommissionNode(t *testing.T, sb testenv.DiffingSandbox, b ClusterB
 			t.Log("statefulset replicas do not match")
 			return false, nil
 		}
+		err = makeDrainStatusChecker(t, sb, b, uint64(numNodes))
+		if err != nil {
+			t.Logf("makeDrainStatusChecker failed due to error %v\n", err)
+			logPods(context.TODO(), sts, cluster, sb, t)
+			return false, err
+		}
+		return true, nil
+	})
+	require.NoError(t, err)
+}
+
+// RequireDecommisionDrainStatusNode checks that proper nodes are decommisioned using node status cmd
+func RequireDecommisionDrainStatusNode(t *testing.T, sb testenv.DiffingSandbox, b ClusterBuilder, numNodes int32) {
+	err := wait.Poll(10*time.Second, 400*time.Second, func() (bool, error) {
+		err := makeDrainStatusChecker(t, sb, b, uint64(numNodes))
+		if err != nil {
+			t.Logf("makeDrainStatusChecker failed due to error %v\n", err)
+			return false, err
+		}
 		return true, nil
 	})
 	require.NoError(t, err)
 }
 
+func makeDrainStatusChecker(t *testing.T, sb testenv.DiffingSandbox, b ClusterBuilder, numNodes uint64) error {
+	cluster := b.Cluster()
+	cmd := []string{"/cockroach/cockroach", "node", "status", "--decommission", "--format=csv", cluster.SecureMode(), fmt.Sprintf("--port=%d", *cluster.Spec().GRPCPort)}
+
+	stdout, stderror, err := kube.ExecInPod(sb.Mgr.GetScheme(), sb.Mgr.GetConfig(), cluster.Namespace(),
+		fmt.Sprintf("%s-0", cluster.StatefulSetName()), resource.DbContainerName, cmd)
+	if err != nil {
+		t.Logf("exec cmd = %s on pod= %s-0 exit with error %v and stdError %s", cmd, cluster.StatefulSetName(), err, stderror)
+		return err
+	}
+	if stderror != "" {
+		t.Logf("exec cmd = %s on pod= %s-0 exit with error %v and stdError %s", cmd, cluster.StatefulSetName(), err, stderror)
+		return err
```

Comment:
This will always be `nil` maybe we need to create a new error specific to this?

Reply:
stderror is a string and contains the stderror content from the command we exec in bash. In case of a failure of the bash command we can debug better.

Reply:
Right, but just above we check `err != nil` and return the error, so the `stderror` check is only reached if `err == nil` and then we return `nil` so from the calling functions perspective the function exited successfully even though we exited out early.